### DISCUSSION TITLE
curl ${CURL_FLAGS} can contain 1 flag but not several

### DISCUSSION THREAD

**Discussion by pheiduck** [curl/curl#14865] [Q&A] [answered]
I saved some curl options into a variable like CURL_FLAGS.:
`CURL_FLAGS=--ping --pong`
During the curl execution `curl ${CURL_FLAGS} https://example.com/text.txt` curl throw an error:
```
curl: option --compressed --progress-bar: is unknown
```

**curl 7.68.0**

**Comment by dfandrich** [comment, marked as answer]
This looks like an issue with the use of the shell. `CURL_FLAGS=--ping --pong` causes an error `bash: --pong: command not found` because the two arguments aren't quoted and the second is treated as a command. If they're quoted, it works fine. e.g. `CURL_FLAGS='--help --variable' && curl ${CURL_FLAGS}`  It causes an error more like the one you show if the variable is quoted when accessed instead:
```
$ CURL_FLAGS='--help --variable'
$ curl "${CURL_FLAGS}"
curl: option --help --variable: is unknown
```
because it's interpreted as a single option that contains a space in the middle.
